services/sequelize/recovery_method.js

```diff
@@ -0,0 +1,169 @@
+const { Op } = require('sequelize');
+
+const models = require('../../models');
+
+const { Account } = models;
+
+const SequelizeRecoveryMethods = {
+    cleanRecoveryMethod(recoveryMethod) {
+        if (!recoveryMethod) {
+            return null;
+        }
+
+        const {
+            kind,
+            detail,
+            publicKey,
+            requestId,
+            securityCode,
+            createdAt,
+            updatedAt,
+        } = recoveryMethod.toJSON();
+
+        return {
+            kind,
+            detail,
+            publicKey,
+            requestId: parseInt(requestId, 10),
+            securityCode,
+            createdAt,
+            updatedAt,
+        };
+    },
+
+    async createRecoveryMethod({ accountId, detail, kind, publicKey, requestId, securityCode }) {
+        const account = await Account.findOne({ where: { accountId } });
+        const recoveryMethod = await account.createRecoveryMethod({
+            kind,
+            ...(detail && { detail }),
+            ...(publicKey && { publicKey }),
+            ...(requestId && { requestId }),
+            ...(securityCode && { securityCode }),
+        });
+
+        return this.cleanRecoveryMethod(recoveryMethod);
+    },
+
+    async deleteOtherRecoveryMethods({ accountId, detail }) {
+        const account = await Account.findOne({ where: { accountId } });
+        if (!account) {
+            return null;
+        }
+
+        const allRecoveryMethods = await account.getRecoveryMethods();
```

Comment:
When we port to DDB, we should make these deletes run in parallel :)

Reply:
😎